service/src/main/java/bio/terra/workspace/db/ResourceDao.java

```diff
@@ -517,31 +531,31 @@ public void createControlledResource(ControlledResource controlledResource)
     // uniqueness rules. This prevents a race condition allowing a new resource to point to the same
     // cloud artifact as another, even if it has a different resource name and ID.
     switch (controlledResource.getResourceType()) {
-      case GCS_BUCKET:
+      case CONTROLLED_GCP_GCS_BUCKET:
         validateUniqueGcsBucket(controlledResource.castToGcsBucketResource());
```

Comment:
Can the validate function live on the enum so you can kill anoteh `switch`? You would also need to put a method reference to the casting function there. But then it would just be `controlledResource.getResourceType().validate()` where `validate` would know to call the right cast function.

Reply:
Yes, that is the plan. I will do that in another PR.